### PR TITLE
Remove boost::filesystem from builds

### PR DESCRIPTION
**Pull request by rmccorm4**
Server: https://github.com/triton-inference-server/server/pull/6810
Core: https://github.com/triton-inference-server/core/pull/319

Fixes: https://github.com/triton-inference-server/server/issues/6745

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -974,7 +974,6 @@ def create_dockerfile_buildbase(ddir, dockerfile_name, argmap):
 RUN wget -O /tmp/boost.tar.gz \
         https://archives.boost.io/release/1.80.0/source/boost_1_80_0.tar.gz && \
     (cd /tmp && tar xzf boost.tar.gz) && \
-    cd /tmp/boost_1_80_0 && ./bootstrap.sh --prefix=/usr && ./b2 install && \
```
Comment (oandreeva-nv) [inline build.py:977]: May I ask you to briefly clarify where else `boost` is used and why we are keeping it here?

Comment (rmccorm4) [reply build.py:977]: We use the header-only boost libs (`span` in cache, `interprocess_lock` in python backend, `stacktrace` in server, and some others) in several places. `boost::filesystem` was the only one that needed a runtime shared library requiring this extra `bootstrap+install` step.

Comment (oandreeva-nv) [reply build.py:977]: Got it, thanks

**Review by oandreeva-nv — APPROVED** [review]

**Comment by rmccorm4** [comment]
Passed L0_model_config and the relevant `test_file_override_security` subtest of L0_lifecycle.

File: build.py (hunk 1 above)
Comment (vadimkantorov) [reply build.py:977]: Is there still an external dependency on libboost? Or could these header-only libs be included in-tree?

**Comment by vadimkantorov** [comment]
boost is now not required, right? starting from what triton release?

thanks!

**Pull request merged**